[Bug]: Getting `interface conversion: interface {} is nil, not map[string]interface {}` on `aws_glue_connection` when `physical_connection_requirements` is `{}`

### Terraform Core Version

1.6.5

### AWS Provider Version

5.29.0

### Affected Resource(s)

`aws_glue_connection`

### Expected Behavior

Providing an `{}` `physical_connection_requirements` should ignore it. 

### Actual Behavior

I get this error

```
Stack trace from the terraform-provider-aws_v5.29.0_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 122 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/glue.expandConnectionInput(0x7f9a272acad8?)
github.com/hashicorp/terraform-provider-aws/internal/service/glue/connection.go:282 +0x5aa
github.com/hashicorp/terraform-provider-aws/internal/service/glue.resourceConnectionCreate({0xf7cf580, 0xc002833800}, 0x0?, {0xe2d60e0?, 0xc00024cb40?})
github.com/hashicorp/terraform-provider-aws/internal/service/glue/connection.go:128 +0x209
github.com/hashicorp/terraform-provider-aws/internal/provider.interceptedHandler[...].func1(0x0?, {0xe2d60e0?, 0xc00024cb40?})
github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:111 +0x32d
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xf7cf580?, {0xf7cf580?, 0xc00256fbf0?}, 0xd?, {0xe2d60e0?, 0xc00024cb40?})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.30.0/helper/schema/resource.go:773 +0x87
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0014faa80, {0xf7cf580, 0xc00256fbf0}, 0xc00255be10, 0xc000e21480, {0xe2d60e0, 0xc00024cb40})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.30.0/helper/schema/resource.go:909 +0xa7e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0017c3de8, {0xf7cf580?, 0xc00256fad0?}, 0xc000badf40)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.30.0/helper/schema/grpc_provider.go:1072 +0xe8d
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0xf7cf4d8?, {0xf7cf580?, 0xc00256f7a0?}, 0xc000badf40)
github.com/hashicorp/terraform-plugin-mux@v0.12.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x1b5
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc003926000, {0xf7cf580?, 0xc00256ef90?}, 0xc0009ebb20)
github.com/hashicorp/terraform-plugin-go@v0.19.1/tfprotov5/tf5server/server.go:859 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xe043d80?, 0xc003926000}, {0xf7cf580, 0xc00256ef90}, 0xc000e21200, 0x0)
github.com/hashicorp/terraform-plugin-go@v0.19.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:467 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0004c72c0, {0xf7cf580, 0xc00256eed0}, {0xf7df1a0, 0xc000ba61a0}, 0xc0025699e0, 0xc00391d800, 0x15fd0d38, 0x0)
google.golang.org/grpc@v1.59.0/server.go:1343 +0xe49
google.golang.org/grpc.(*Server).handleStream(0xc0004c72c0, {0xf7df1a0, 0xc000ba61a0}, 0xc0025699e0)
google.golang.org/grpc@v1.59.0/server.go:1737 +0xca6
google.golang.org/grpc.(*Server).serveStreams.func1.1()
google.golang.org/grpc@v1.59.0/server.go:986 +0x8c
created by google.golang.org/grpc.(*Server).serveStreams.func1
google.golang.org/grpc@v1.59.0/server.go:997 +0x15c

Error: The terraform-provider-aws_v5.29.0_x5 plugin crashed!
```

### Relevant Error/Panic Output Snippet

_No response_

### Terraform Configuration Files

```
resource "aws_glue_connection" "default" {
  for_each = length(var.glue_connection) != 0 ? { for v in var.glue_connection : v.name_suffix => v } : {}

  name                  = "${local.name_prefix}-${each.value.name_suffix}-${data.aws_region.current.name}"
  connection_properties = each.value.connection_properties
  connection_type       = each.value.type
  description           = each.value.description
  tags                  = var.tags

  physical_connection_requirements {
    availability_zone      = each.value.availability_zone
    security_group_id_list = each.value.security_groups
    subnet_id              = each.value.subnet_id
  }
}

variable "glue_connection" {
  type = list(object({
    availability_zone     = optional(string)
    description           = string
    connection_properties = optional(map(string), {})
    name_suffix           = string
    security_groups       = optional(list(string), [])
    subnet_id             = optional(string)
    type                  = optional(string, "JDBC")
  }))
  default     = []
  description = "List of glue connection details"

  validation {
    condition     = alltrue([for v in var.glue_connection : contains(["CUSTOM", "JDBC", "KAFKA", "MARKETPLACE", "MONGODB", "NETWORK"], v.type)])
    error_message = "Supported Glue connection types are: CUSTOM, JDBC, KAFKA, MARKETPLACE, MONGODB, and NETWORK."
  }
}

glue_connection = [{
    description = "CB Global redshift connection"
    name_suffix = "cbglobal_${var.pipeline_setup.type}"

    connection_properties = {
      JDBC_CONNECTION_URL = "***"
      PASSWORD            = var.password
      USERNAME            = var.username
    }
  }]
```

OR

```
resource "aws_glue_connection" "default" {
  name  = "test"

  physical_connection_requirements {
   
  }
}
```

### Steps to Reproduce

```
terrafom innit
terraform apply
```

### Debug Output

```
Stack trace from the terraform-provider-aws_v5.29.0_x5 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 122 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/glue.expandConnectionInput(0x7f9a272acad8?)
github.com/hashicorp/terraform-provider-aws/internal/service/glue/connection.go:282 +0x5aa
github.com/hashicorp/terraform-provider-aws/internal/service/glue.resourceConnectionCreate({0xf7cf580, 0xc002833800}, 0x0?, {0xe2d60e0?, 0xc00024cb40?})
github.com/hashicorp/terraform-provider-aws/internal/service/glue/connection.go:128 +0x209
github.com/hashicorp/terraform-provider-aws/internal/provider.interceptedHandler[...].func1(0x0?, {0xe2d60e0?, 0xc00024cb40?})
github.com/hashicorp/terraform-provider-aws/internal/provider/intercept.go:111 +0x32d
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xf7cf580?, {0xf7cf580?, 0xc00256fbf0?}, 0xd?, {0xe2d60e0?, 0xc00024cb40?})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.30.0/helper/schema/resource.go:773 +0x87
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0014faa80, {0xf7cf580, 0xc00256fbf0}, 0xc00255be10, 0xc000e21480, {0xe2d60e0, 0xc00024cb40})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.30.0/helper/schema/resource.go:909 +0xa7e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0017c3de8, {0xf7cf580?, 0xc00256fad0?}, 0xc000badf40)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.30.0/helper/schema/grpc_provider.go:1072 +0xe8d
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0xf7cf4d8?, {0xf7cf580?, 0xc00256f7a0?}, 0xc000badf40)
github.com/hashicorp/terraform-plugin-mux@v0.12.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x1b5
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc003926000, {0xf7cf580?, 0xc00256ef90?}, 0xc0009ebb20)
github.com/hashicorp/terraform-plugin-go@v0.19.1/tfprotov5/tf5server/server.go:859 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xe043d80?, 0xc003926000}, {0xf7cf580, 0xc00256ef90}, 0xc000e21200, 0x0)
github.com/hashicorp/terraform-plugin-go@v0.19.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:467 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0004c72c0, {0xf7cf580, 0xc00256eed0}, {0xf7df1a0, 0xc000ba61a0}, 0xc0025699e0, 0xc00391d800, 0x15fd0d38, 0x0)
google.golang.org/grpc@v1.59.0/server.go:1343 +0xe49
google.golang.org/grpc.(*Server).handleStream(0xc0004c72c0, {0xf7df1a0, 0xc000ba61a0}, 0xc0025699e0)
google.golang.org/grpc@v1.59.0/server.go:1737 +0xca6
google.golang.org/grpc.(*Server).serveStreams.func1.1()
google.golang.org/grpc@v1.59.0/server.go:986 +0x8c
created by google.golang.org/grpc.(*Server).serveStreams.func1
google.golang.org/grpc@v1.59.0/server.go:997 +0x15c

Error: The terraform-provider-aws_v5.29.0_x5 plugin crashed!
```

### Panic Output

_No response_

### Important Factoids

_No response_

### References

_No response_

### Would you like to implement a fix?

Yes

--- a/.changelog/34737.txt
+++ b/.changelog/34737.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_glue_connection: Fix Terraform crash while creating resource with empty `physical_connection_requirements` configuration block.
+```

--- a/.changelog/34737.txt
+++ b/.changelog/34737.txt
@@ -1,3 +1,3 @@
 ```release-note:bug
-resource/aws_glue_connection: Fix Terraform crash while creating resource with empty `physical_connection_requirements` configuration block.
+resource/aws_glue_connection: Fix crash while creating resource with empty `physical_connection_requirements` configuration block
 ```

--- a/internal/service/glue/connection.go
+++ b/internal/service/glue/connection.go
@@ -277,9 +277,8 @@ func expandConnectionInput(d *schema.ResourceData) *glue.ConnectionInput {
 		connectionInput.MatchCriteria = flex.ExpandStringList(v.([]interface{}))
 	}
 
-	if v, ok := d.GetOk("physical_connection_requirements"); ok {
-		physicalConnectionRequirementsList := v.([]interface{})
-		physicalConnectionRequirementsMap := physicalConnectionRequirementsList[0].(map[string]interface{})
+	if v, ok := d.GetOk("physical_connection_requirements"); ok && v.([]interface{})[0] != nil {
+		physicalConnectionRequirementsMap := v.([]interface{})[0].(map[string]interface{})
 		connectionInput.PhysicalConnectionRequirements = expandPhysicalConnectionRequirements(physicalConnectionRequirementsMap)
 	}
 

